grails-app/services/org/pih/warehouse/requisition/RequisitionService.groovy

```diff
@@ -825,23 +825,34 @@ class RequisitionService {
         return new Event(eventDate: eventDate, eventType: eventType, eventLocation: eventLocation, createdBy: currentUser)
     }
 
+    Requisition transitionRequisitionStatus(Requisition requisition, RequisitionStatus requisitionStatus, EventCode eventCode, User currentUser) {
+        requisition.status = requisitionStatus
+        Event event = createEvent(eventCode, requisition.origin, new Date(), currentUser)
+        requisition.addToEvents(event)
+        requisition.save()
+    }
+
     void triggerRequisitionStatusTransition(Requisition requisitionInstance, User currentUser) {
         Requisition.withSession {
             Requisition requisition = Requisition.get(requisitionInstance.id)
             if (requisition.status == RequisitionStatus.PENDING_APPROVAL) {
                 if (requisition.origin.approvalRequired) {
-                    Event event = createEvent(EventCode.PENDING_APPROVAL, requisition.origin, new Date(), currentUser)
-                    requisition.addToEvents(event)
-                    requisition.status = RequisitionStatus.PENDING_APPROVAL
+                    transitionRequisitionStatus(requisition, RequisitionStatus.PENDING_APPROVAL, EventCode.PENDING_APPROVAL, currentUser)
                     requisition.approvalRequired = true
-                    requisition.save()
                     return
                 }
-                Event event = createEvent(EventCode.SUBMITTED, requisition.origin, new Date(), currentUser)
-                requisition.addToEvents(event)
-                requisition.status = RequisitionStatus.VERIFYING
+                transitionRequisitionStatus(requisition, RequisitionStatus.VERIFYING, EventCode.SUBMITTED, currentUser)
                 requisition.approvalRequired = false
-                requisition.save()
+            } else if (requisition.status == RequisitionStatus.APPROVED) {
+                transitionRequisitionStatus(requisition, RequisitionStatus.APPROVED, EventCode.APPROVED, currentUser)
+                requisition.dateApproved = new Date()
+                requisition.approvedBy = currentUser
+                return
+            } else if (requisition.status == RequisitionStatus.REJECTED) {
```

Comment:
nitpicky: `else` is not required if you call `return` above (take a look at my suggestion below)

Reply:
I'd rather remove the return than the else.